lib/husky_musher/utils/redcap.py

```diff
@@ -0,0 +1,529 @@
+    *event* of the *record_id*.
+
+    Will include the repeat *instance* if provided.
+    """
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'surveyLink',
+        'format': 'json',
+        'instrument': instrument,
+        'event': event,
+        'record': record_id,
+        'returnFormat': 'json'
+    }
+
+    if instance:
+        data['repeat_instance'] = str(instance)
+
+    response = requests.post(REDCAP_API_URL, data=data)
+    response.raise_for_status()
+    return response.text
+
+
+def get_todays_repeat_instance() -> int:
+    """
+    Returns the repeat instance number, i.e. days since the start of the study
+    with the first instance starting at 1.
+    """
+    return 1 + (datetime.today() - STUDY_START_DATE).days
+
+
+def redcap_registration_complete(redcap_record: dict) -> bool:
+    """
+    Returns True if a given *redcap_record* shows a participant has completed
+    the enrollment surveys. Otherwise, returns False.
+
+    >>> redcap_registration_complete(None)
+    False
+
+    >>> redcap_registration_complete({})
+    False
+
+    >>> redcap_registration_complete({ \
+        'eligibility_screening_complete': '1', \
+        'consent_form_complete': '2', \
+        'enrollment_questionnaire_complete': '0'})
+    False
+
+    >>> redcap_registration_complete({ \
+        'eligibility_screening_complete': '2', \
+        'consent_form_complete': '2', \
+        'enrollment_questionnaire_complete': '1'})
+    False
+
+    >>> redcap_registration_complete({ \
+        'eligibility_screening_complete': '2', \
+        'consent_form_complete': '2', \
+        'enrollment_questionnaire_complete': '2'})
+    True
+    """
+    if not redcap_record:
+        return False
+
+    return (is_complete('eligibility_screening', redcap_record) and \
+            is_complete('consent_form', redcap_record) and \
+            is_complete('enrollment_questionnaire', redcap_record))
+
+
+def fetch_encounter_events_past_week(redcap_record: dict) -> List[dict]:
+    """
+    Given a *redcap_record*, export the full list of related REDCap instances
+    from the Encounter arm of the project that have occurred in the past week.
+    """
+    # Unfortunately, despite its appearance in the returned response from REDCap,
+    # `redcap_repeat_instance` is not a field we can query by when exporting
+    # REDCap records. Thus, return all information from a REDCap record (without
+    # specifying fields) to retrieve the instance key.
+    data = {
+        'token': REDCAP_API_TOKEN,
+        'content': 'record',
+        'format': 'json',
+        'type': 'flat',
+        'csvDelimiter': '',
+        'events': 'encounter_arm_1',
+        'filterLogic': f'[record_id] = "{redcap_record["record_id"]}"',
+        'rawOrLabel': 'label',
+        'rawOrLabelHeaders': 'raw',
+        'exportCheckboxLabel': 'false',
+        'exportSurveyFields': 'false',
+        'exportDataAccessGroups': 'false',
+        'returnFormat': 'json'
```

Comment:
Ah yes, I remember now. All instances are still returned (regardless of modification date) because they're all still attached to the same modified record.

Reply:
Booo, that's too bad. Thank you for testing it out.